include/simdjson/generic/ondemand/document.h

```diff
@@ -198,7 +198,11 @@ class document {
    */
   simdjson_really_inline operator bool() noexcept(false);
 #endif
-
+  /**
+  * This method scans the array and counts the number of elements.
+  * The runtime complexity is linear in the size of the array.
+  */
+  simdjson_really_inline simdjson_result<size_t> count_elements() & noexcept;
```

Comment:
If we're doing this on document, we should do it on value too. (I think it's perfectly reasonable to do it on document and value, btw!)

Reply:
Yep.